### DISCUSSION TITLE
When Failed to connect to relay server, show server IP

### DISCUSSION THREAD

**Discussion by loliconRoot** [rustdesk/rustdesk#3823] [Ideas]
### Description

When connecting to a remote host, the relay settings in its rustdesk will override the client's relay server settings in a silent state, which can lead to some very confusing behavior such as: if the ID server and the relay server settings are the same, you can connect to the ID server normally, but you cannot connect to the relay server. A more extreme case is to set the rustdesk relay server where the server is located to 127.0.0.1. This seems to cause any other client in the network to be able to search whether the remote host is connected to the network but not control the remote host.
Although this is a very unusual situation, the silent behavior can make it confusing and difficult to debug the results.

### Impact

Custom relay servers users

**Comment by rudolfbyker** [comment]
To me, it makes no sense that the server's `-r` setting (i.e. the advertised relay address) should override the client's "Relay server" setting. It would make much more sense if the server advertised the default, and the client could override that if they wanted to.
